source/client-side-encryption/client-side-encryption.rst

```diff
@@ -914,6 +912,17 @@ Drivers MUST support a BSON document option named ``encryptedFields`` for any
 will be interpreted by the helper method and MUST be passed to the `create`_
 command.
 
+.. note::
+   Users are not expected to set the ``escCollection`` and ``ecocCollection`` in
+   ``encryptedFields``. SERVER-74069 added server-side validation for those fields
+   and no longer allows names to deviate from the following:
```

Comment:
Does it make sense for the `encryptedFieldsMap` in the test files to still define `escCollection` and `ecocCollection`? I originally assumed [DRIVERS-2586](https://jira.mongodb.org/browse/DRIVERS-2586) would remove the options entirely. Drivers would never consult the options and just infer names for the metadata collections using the spec format. We have CommandStartedEvent expectations for creating the metadata collections so that's covered. For the encrypted collection's CommandStartedEvent, we could then assert that the metadata names are _not_ included in the `encryptedFields` option.

This relates to my later comment in `fle2v2-CreateCollection.yml` about adding `encryptedFields` to the `create` command's CommandStartedEvent and asserting that `escCollection` and `ecocCollection` are left unset.

----

As far as suggested spec language changes, I see two alternatives:

### Drivers consult the metadata collection options

Add language that drivers MUST consult the options if specified and use them to infer the metadata collections to create.

Add tests where metadata collections with invalid names are created (using CommandStartedEvent expectations and/or assertCollectionExists after the fact). The test should expect the `createCollection()` operation to fail due to a server-side error. The CommandStartedEvent for the encrypted collection's `create` command (which fails) should assert that the `encryptedFields` option includes the invalid metadata names.

### Drivers ignore the metadata collection options

Add language that drivers MUST ignore the options. They will not be used to derive names of metadata collections to create or drop.

Add a test where invalid metadata collection names are specified in `encryptedFields` (either the client-level map or operation-level option). Assert that metadata collections with the correct names are created (thus the options are ignored). Assert that `createCollection()` fails due to a server-side error since the driver leaves those options as-is

Reply:
> Does it make sense for the `encryptedFieldsMap` in the test files to still define `escCollection` and `ecocCollection`?

Good question. I think "no" since users are not expected to set `escCollection` or `ecocCollection`. https://github.com/mongodb/specifications/pull/1396/commits/f7c028a86779ba2a073af9f0df72c5c2d1bc0f6c removes `escCollection` and `ecocCollection` from tests.

> I originally assumed [DRIVERS-2586](https://jira.mongodb.org/browse/DRIVERS-2586) would remove the options entirely. Drivers would never consult the options and just infer names for the metadata collections using the spec format. We have CommandStartedEvent expectations for creating the metadata collections so that's covered. For the encrypted collection's CommandStartedEvent, we could then assert that the metadata names are _not_ included in the `encryptedFields` option.

426533fc0bc373128c4d9527342ae25cbf2afec5 adds `{$$exists: false}` for metadata name fields in `create` command sent to server.


> As far as suggested spec language changes, I see two alternatives:

I chose to keep the existing behavior: first consult `encryptedFields` for the metadata collection names. Ignoring `encryptedFields` may result in a mismatch of the metadata collection names for the collections the driver creates, and the names referenced in `encryptedFields` sent to the server. If the server ever relaxes restrictions added in SERVER-74069, the `createCollection` may operation succeed.

> 
> ### Drivers consult the metadata collection options
> Add language that drivers MUST consult the options if specified and use them to infer the metadata collections to create.

`Create Collection Helper` already requires consulting `encryptedFields` first for the metadata collection names. No change was made.


> Add tests where metadata collections with invalid names are created (using CommandStartedEvent expectations and/or assertCollectionExists after the fact). The test should expect the `createCollection()` operation to fail due to a server-side error. The CommandStartedEvent for the encrypted collection's `create` command (which fails) should assert that the `encryptedFields` option includes the invalid metadata names.

I like the idea. fbc7b17e2da1282a430f7f18a8ef49db48540e1a adds a test.

Reply:
I followed up on the `$$exists` change in https://github.com/mongodb/specifications/pull/1396/files#r1161553002.

Reply:
> I chose to keep the existing behavior: first consult encryptedFields for the metadata collection names. Ignoring encryptedFields may result in a mismatch of the metadata collection names for the collections the driver creates, and the names referenced in encryptedFields sent to the server.

For drivers where encryptedFields is an arbitrary BSON document, I can see the logic in still consulting the state collection names if present.

For strongly typed languages that may be modeling encryptedFields, wouldn't this PR suggest they remove the options entirely from their public API entirely? Otherwise, having the options in a public API might as well be documentation (the setters could show up in code completion).

Just curious if that's something that still needs to be addressed here.

Reply:
> For strongly typed languages that may be modeling encryptedFields

I expect no drivers are modeling encryptedFields. [Create Collection Helper](https://github.com/mongodb/specifications/blob/fcbd905969a1178ee979d426be4dc73437e96079/source/client-side-encryption/client-side-encryption.rst#create-collection-helper) notes:
> Drivers MUST support a BSON document option named `encryptedFields`

[This survey](https://docs.google.com/spreadsheets/d/10EGNGOR4b5vyRuVI3VryN3Fl18Xe660OEYgVTYFCDYY/edit#gid=0) suggests no drivers are modeling `encryptedFields`.